### PR TITLE
feat: add duration, currentTime, file url, oncomplete

### PR DESCRIPTION
**Pull request by kheftel**
Added some new features, fixed some bugs. Rebased it on your most recent 0.1.3 release. Hope it's not too much for one PR. Implemented and tested on iOS and Android.

Added the ability to get the duration of a loaded audio file. Restricted it to only `preloadComplex` sounds with 1 channel for simplicity.
Added the ability to get the current time of a playing audio file for progress listeners / updates. Restricted it to only `preloadComplex` sounds with 1 channel for simplicity.
Added a complete event at the end of a (complex) playing audio file.
Added the ability to ask if a particular audio is loaded (since the `unload` event now throws an error on android if a file is already loaded).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,18 +115,20 @@ NativeAudio.preloadSimple({
 /**
  * Platform: Android/iOS
  * This method will load more optimized audio files for background into memory.
- * @param assetPath - relative path of the file or absolute url (http://)
+ * @param assetPath - relative path of the file or absolute url (file://)
```
Comment (Boosten) [inline README.md:118]: Good catch. I was in the impression that I could use this plugin with external resources too. 
The preloadSimple comment still contains "http://" though. 

Maybe even add a remark to the readme that this plugin only works with local files?

Comment (kheftel) [reply README.md:118]: sure, I can do that. 

actually I'm pretty sure that before this PR, you could only load assets from the app bundle / resources, as I tried to use `file` URLs and failed, and thus added the `isUrl` parameter to get them to work (but only for `preloadComplex`).

My use case is downloading audio files to the app cache directory and playing them from there, hence I added support for `file` URLs.

Comment (kheftel) [reply README.md]: I added that second commit, let me know what else I can do.

Comment (Boosten) [reply README.md]: Looks good to me 👍  Now we wait for someone to approve this PR 😉

Comment (kheftel) [reply README.md]: sounds good!

**Comment by kheftel** [comment]
I added a documentation change to this PR to clarify, per #9, that on Android, your asset file must be in `res/raw` and you must pass a filename w/o extension for preloading to work properly.

**Comment by yacut** [comment]
@Boosten @priyankpat Any update on this?

**Comment by Boosten** [comment]
> @Boosten @priyankpat Any update on this?

@priyankpat has to approve and merge the PR :-)

**Comment by damngamerz** [comment]
This is really cool can we merge these changes @priyankpat

**Comment by danielehrhardt** [comment]
Please merge

**Comment by taylorchance** [comment]
Please merge, I pulled his repo locally and it worked perfectly for me. This is a must needed feature and there aren't many other options.

**Comment by HunSpeedi** [comment]
Also leaving a tactical dot for a merge request: .
Thanks in advance @priyankpat & @kheftel :)

**Comment by kheftel** [comment]
@priyankpat if you add me as a maintainer I'll get this merged.

**Comment by kheftel** [comment]
@thomasvidas, I'm not, but am joining now.

**Comment by bazuka5801** [comment]
All these changes add in new version!

**Pull request closed without merging**